internal/dataplane/parser/atc/predicate.go

```diff
@@ -115,15 +116,36 @@ func (p Predicate) IsEmpty() bool {
 	return p.value == nil
 }
 
+var (
+	ErrorTypeNotMatch    = errors.New("type does not match on sides of predicate")
+	ErrorOperatorInvalid = errors.New("operator is not valid for the types of sides of predicate")
```

Comment:
These should be called `Err...` per the [convention](https://github.com/golang/go/wiki/Errors#naming).

I've introduced the `errname` linter to enforce this https://github.com/Kong/kubernetes-ingress-controller/pull/4811

Reply:
updated